src/github.com/couchbase/sync_gateway/base/bucket_gocb.go

```diff
@@ -82,12 +92,26 @@ func (bucket CouchbaseBucketGoCB) GetName() string {
 }
 
 func (bucket CouchbaseBucketGoCB) Get(k string, rv interface{}) (cas uint64, err error) {
+	bucket.singleOps <- struct{}{}
+	gocbExpvars.Add("SingleOps", 1)
+	defer func() {
+		<-bucket.singleOps
+		gocbExpvars.Add("SingleOps", -1)
+	}()
 	gocbExpvars.Add("Get", 1)
 	casGoCB, err := bucket.Bucket.Get(k, rv)
 	return uint64(casGoCB), err
 }
 
 func (bucket CouchbaseBucketGoCB) GetRaw(k string) (rv []byte, cas uint64, err error) {
+
+	bucket.singleOps <- struct{}{}
```

Comment:
Can you add a comment above this block that says something like "Limit number of concurrent GoCB calls in order to avoid queue overflow issues"